azure-pipelines/ui-automation/msal-broker-release.yml

```diff
@@ -0,0 +1,147 @@
+    branches:
+      include:
+        - dev
+    always: true
+
+parameters:
+  - name: firebaseDeviceId
+    type: string
+    displayName: Firebase Device Id
+    default: blueline
+    values:
+      - blueline
+      - flame
+      - redfin
+      - sailfish
+      - walleye
+      - bluejay
+      - oriole
+  - name: firebaseDeviceAndroidVersion
+    displayName: Firebase Device Android Version
+    type: number
+    default: 28
+    values:
+      - 25
+      - 26
+      - 27
+      - 28
+      - 29
+      - 30
+      - 31
+      - 32
+      - 33
+  - name: testTargetPackages
+    displayName: Packages as Test Targets
+    type: string
+    default: package com.microsoft.identity.client.msal.automationapp.testpass.broker
+  - name: oldBrokerHostVersion
+    displayName: Old Broker host Version
+    type: string
+    default: '0.0.1'
+  - name: msalVersion
+    displayName: MSAL Version
+    type: string
+
+variables:
+  engineeringProjectId: 'fac9d424-53d2-45c0-91b5-ef6ba7a6bf26'
+  azureSamplePipelineId: 1458
+  brokerHostPipelineId: 1432
+  msazureServiceConnection: AndroidBroker-CI
+  msazureFeedName: Android-Broker
+  azureSampleApk: AzureSample-local-debug.apk
+  brokerHostApk: brokerHost-local-debug.apk
+  oldBrokerHostApk: brokerHost-local-debug.apk
+  firebaseTimeout: 45m
+
+stages:
+  # msalautomationapp
+  - stage: 'msalautomationapp'
+    displayName: Build MSAL Automation APKs
+    jobs:
+      - template: ./templates/build-msal-automation-app.yml
+        parameters:
+          brokerFlavor: AutoBroker
+          msalFlavor: Dist
+          brokerSource: PlayStore
+          msalVersion: ${{ parameters.msalVersion }}
+  # Brokers
+  - stage: 'brokers'
+    dependsOn: []    # this removes the implicit dependency on previous stage and causes this to run in parallel
+    displayName: Brokers and Azure Sample APKs
+    jobs:
+      - job: 'download_brokers'
+        displayName: Download Brokers
+        pool:
+          vmImage: ubuntu-latest
+        steps:
+          - checkout: none
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download latest Azure Sample'
+            inputs:
+              buildType: 'specific'
+              project: '$(engineeringProjectId)'
+              definition: '$(azureSamplePipelineId)'
+              artifactName: AzureSample
+              itemPattern: '**/*.apk'
+              targetPath: '$(Build.ArtifactStagingDirectory)/azureSample'
+              buildVersionToDownload: 'latest'
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download Broker Host'
+            inputs:
+              buildType: specific
+              project: '$(engineeringProjectId)'
+              definition: '$(brokerHostPipelineId)'
+              artifactName: BrokerHost
+              itemPattern: '**/*.apk'
+              targetPath: '$(Build.ArtifactStagingDirectory)/brokerHost'
+          - task: UniversalPackages@0
```

Comment:
I think you will also need to download an Old Authenticator for the 4 update scenario at pop tests.

I'm using this in another PR:
- task: UniversalPackages@0
      displayName: 'Download old authenticator (Pre v5 Broker) version from feed'
      inputs:
        command: 'download'
        downloadDirectory: '$(Build.ArtifactStagingDirectory)/brokers/oldAPKs'
        feedsToUse: 'internal'
        vstsFeed: '${{ parameters.internalFeedName }}'
        vstsFeedPackage: 'com.azure.authenticator'
        vstsPackageVersion: '${{ parameters.oldAuthenticatorVersion }}'
        
where  oldAuthenticatorVersion is a parameter with default value 6.2204.2470
and internalFeedName is AndroidAdal

Reply:
Also I'm not sure you need to download the old brokerhost at all

Reply:
This is running against prod Authenticator and CP, right?

Reply:
This is running against auth app, cp yes